metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -286,16 +309,57 @@ class TableauConfig(
         description="Ingest a URL to render an embedded Preview of assets within Tableau.",
     )
 
+    extract_project_hierarchy: bool = Field(
+        default=True,
+        description="Whether to extract entire project hierarchy for nested projects.",
+    )
+
+    @root_validator(pre=False)
+    def projects_backward_compatibility(cls, values: Dict) -> Dict:
+        projects = values.get("projects")
+        project_pattern = values.get("project_pattern")
+        if project_pattern == AllowDenyPattern.allow_all() and projects:
```

Comment:
Here's outcome I want:

For new ingestions, by default we ingest as much data as we can from TABLEAU. This means we ingest ALL PROJECTS unless someone opts out. 

projects: ["a", "b", "c"]

in existing recipe, then this should continue to work.

if projects is NOT set (defaults to [ "default" ]), then upgrade to allow_all.

Reply:
2 things we need to do:

1. add pydantic validator on config dict. if projects field was explicitly provided by user without defaults, then always accept it. if not, always fallback to ALLOW ALL project pattern.
2. add projects as a "Deprecated field" using our helper pydantic_field_deprecated

Reply:
Now tableau source will ingest all project if projects and project_pattern is not provided